### PR TITLE
USF-1440: Checkout feature matrix

### PR DESCRIPTION
**Pull request by jeff-matthews**
This pull request (PR) adds a table showing which Adobe Commerce features that the checkout dropin supports.

It also hides incomplete checkout content. We should avoid publishing blank pages in the future.

>[!NOTE]
>The `OptionsTable` component doesn't currently support image links. It would be nice to have icons/.svg files for each status similar to [this](https://developer.adobe.com/commerce/pwa-studio/integrations/adobe-commerce/features/).

I know that we're attempting to finalize the dropin doc template (#41), but since we'll have to refactor the cart docs after that anyway, it seems prudent to get this info out on production. It's unclear how long it will take to get the template finalized.

### CODE REVIEW

File: src/content/docs/dropins/checkout/index.mdx [hunk 1]
```diff
@@ -7,14 +7,46 @@ sidebar:
...
 
 import Aside from '@components/Aside.astro';
+import OptionsTable from '@components/OptionsTable.astro';
 
 The Checkout dropin provides a variety of fully-customizable controls to help complete a purchase. These controls include forms to introduce required information for contact details like email address, delivery and billing addresses, shipping options, and payment methods. Established customers who added items to the cart as a guest have the ability to sign in, automatically loading default addresses and contact details.
 
-<Aside type="wip" title="In Development">
-  The Checkout dropin is in active development. Stay tuned for updates.
-</Aside>
+The following table provides an overview of the Adobe Commerce features that the Checkout dropin supports:
```
Comment (bdenham) [inline src/content/docs/dropins/checkout/index.mdx:16]:
```suggestion
## Commerce feature support

The following table provides an overview of the Adobe Commerce features that the Checkout dropin supports:
```

**Review by bdenham — APPROVED** [review]
I suggested a heading for the table. Otherwise, looks great.

image

**Comment by keharper** [comment]
How about "Supported Commerce features" instead of "Commerce features supported"? That feels more natural to me, but no biggie either way

**Comment by jeff-matthews** [comment]
I converted to the default Starlight table component and added badges. It required some minor style adjustments, which I [made](https://github.com/commerce-docs/microsite-commerce-storefront/commit/f121f7f02a121378a2ffe8f7397e974f505207ab) in a separate integration branch (now the base branch for this PR):

checkout

**Review by bdenham — APPROVED** [review]
I like the use of the Badge components here! They give the statuses a quick visual representation while providing the text for meaning so that the user does not have to guess. Good thinking!

**Comment by jeff-matthews** [comment]
Approved by Olena in Slack

**Pull request merged**